pkg/api/kptfile/v1alpha2/types.go

```diff
@@ -207,7 +215,7 @@ type Function struct {
 
 	// `Config` specifies an inline KRM resource used as the function config.
 	// Config, ConfigPath, and ConfigMap fields are mutually exclusive.
-	Config *yaml.Node `yaml:"config,omitempty"`
+	Config yaml.Node `yaml:"config,omitempty"`
```

Comment:
Adding a note why this can not be a pointer will be helpful given that it's OneOf(config, configPath, configMap) scenario. I remember discussing this on some PR about how yaml.Zero() can be used to test empty YAML input.